src/AppContainer.tsx

```diff
@@ -28,9 +28,10 @@ type ToolBarContentProps = {
     workspaceInfo: WorkspaceInfo
     extensionsConfig: ExtensionsConfig
     websocketStatus: string
+    onReconnect: () => void
 }
 
-const ToolBarContent: FunctionComponent<ToolBarContentProps> = ({ workspaceInfo, extensionsConfig, websocketStatus }) => {
+const ToolBarContent: FunctionComponent<ToolBarContentProps> = ({ workspaceInfo, extensionsConfig, websocketStatus, onReconnect }) => {
```

Comment:
We now have `onReconnect` callback. When the server api is disconnected, a red symbol appears in the toolbar. If the user clicks it, the onReconnect() is called to attempt to reestablish the connection.